content/blogposts/2023/blah.md

```diff
@@ -0,0 +1,138 @@
+
+In contrast, our question is asking for a _highly precise_ answer related to a specific piece of code. Searching the repository's embeddings for _similar_ code is not the context needed by the LLM to make the correct inferrences. For such implementation-specific questions, we need to be able to explore a different relationship within the code graph.
+
+For the same question, we can generate a very good response from Cody by supplying additional code we find traversing the _definition_ edges of the semantically-indexed portion of the code graph. To start the search, we extract symbol names from the code that is currently in the view of the user, such as the current workspace, the user's text selection, open files, etc. These symbol names are translatted into a location in our SCIP-backed code graph. Then, particular edges are followed out to explore the graph. As we traverse the graph in this manner, we recall code that is semantically related to the code in question.
+
+We've already seen Cody infer infer the type of the variable in question. Including the definition (hence the concrete field names) in the LLM context window provides enough information to effectively prevent the spew of probable-but-fallacious field names we got when it had no concrete examples in its working memory.
+
+<a href="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/req-scip2.png" target="_blank">
+    <Figure
+        src="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/req-scip2.png"
+        alt="SCIP-powered Cody giving a precise answer given appropriate context."
+        caption="SCIP-powered Cody giving a precise answer given appropriate context." />
+</a>
+
+Note that we've stopped additional fields from being emitted, but we didn't happen to expand all of the fields that are present. Currently we're only performing single-edge traversals from the source, but plan to extend the search in the future. Including a second-order relationship included in this example may have prompted Cody to expand the fields present in the embedded struct `CompletionRequestParameters`.
+
+### In the land of infinite possibilities...
+
+**Generative AI gonna generate.**
+
+Too many answers are valid when too little context is given to restrict the solution space. This is a special form of garbage-in, garbage-out. In the following example, we ask Cody to describe the output of a particular function when executed. We choose this particular example in which the input question has little to no similarity to the _internal implementation_ of the function to ensure embeddings search alone would not return all of the relevant context.
```

Comment:
> Too many answers are valid when too little context is given to restrict the solution space.


Very good!! Maybe you could even add example of daily-life-with-chat-gpt, where folks also have learned to add qualifying statements like "assume I'm a software engineer" or "assume you don't need to explain to me what HTTP is" to their prompts.